src/microbe_stage/editor/MicrobeEditor.cs

```diff
@@ -1622,6 +1662,44 @@ private void RemoveOrganelleAt(Hex location)
         EnqueueAction(action);
     }
 
+    [DeserializedCallbackAllowed]
+    private void DoOrganelleMoveAction(MicrobeEditorAction action)
+    {
+        var data = (MoveActionData)action.Data;
+        editedMicrobeOrganelles.Remove(data.Organelle);
+        data.Organelle.Position = data.NewLocation;
+        editedMicrobeOrganelles.Add(data.Organelle);
+        data.Organelle.NumberOfTimesMoved++;
+    }
+
+    [DeserializedCallbackAllowed]
+    private void UndoOrganelleMoveAction(MicrobeEditorAction action)
+    {
+        var data = (MoveActionData)action.Data;
+        editedMicrobeOrganelles.Remove(data.Organelle);
+        data.Organelle.Position = data.OldLocation;
+        editedMicrobeOrganelles.Add(data.Organelle);
+        data.Organelle.NumberOfTimesMoved--;
+    }
+
+    private void MoveOrganelle(Hex oldLocation, Hex newLocation)
+    {
+        var organelleHere = editedMicrobeOrganelles.GetOrganelleAt(oldLocation);
+
+        // Make sure organelle doesn't overlap
+        if (editedMicrobeOrganelles.GetOrganelleAt(newLocation) != null)
```

Comment:
As the old organelle is still in the list, this won't allow a position that slightly (or entirely) overlaps the old position, this needs to be rethought through.